### PR TITLE
fix: hydrate active child-lane manifests for status projection

### PR DESCRIPTION
**Pull request by Kbediako**
## Summary
- hydrate active same-parent child-lane manifests before projecting `Child lane reserved before child run startup.` in `CO STATUS`, `co-status --format json`, and `provider-linear-worker-progress`
- keep child-manifest matching metadata-safe on `parent_run_id`, `issue_id`, `issue_identifier`, and `pipeline_id=provider-linear-child-lane`, with a fail-closed fallback when no matching child evidence exists
- add focused regressions for hydrated reservation projection and malformed child-manifest `run_id` handling

## Validation
- `npx vitest run orchestrator/tests/ProviderLinearWorkerRunner.test.ts orchestrator/tests/SelectedRunProjection.test.ts orchestrator/tests/ProviderIssueObservability.test.ts orchestrator/tests/ControlRuntime.test.ts`
- `node scripts/delegation-guard.mjs`
- `node scripts/spec-guard.mjs --dry-run`
- `npm run build`
- `npm run lint`
- `npm run test`
- `npm run docs:check`
- `npm run repo:stewardship`
- `node scripts/diff-budget.mjs`
- `TASK=linear-34c13c4f-5839-4c64-b6c3-c2b30b7d7fe3 NOTES='Goal: CO-210 active child-lane manifest hydration review | Summary: validate fail-closed manifest matching, reserved-placeholder fallback, and selected-run proof refresh on the current uncommitted diff | Risks: malformed run_id fail-closed handling, lineage overmatch, unrelated docs:freshness gate still red' FORCE_CODEX_REVIEW=1 DIFF_BUDGET_OVERRIDE_REASON='linear-34c13c4f-status-hydration-docs-tests-implementation' CODEX_REVIEW_LARGE_SCOPE_OVERRIDE_REASON='CO-210 spans source, focused regressions, and required docs packet mirrors; uncommitted scope is the auditable review unit for this issue.' npm run review -- --uncommitted`
- `npm run pack:smoke`

## Notes
- `npm run docs:freshness` remains red on the unrelated CO-175 rolling-cohort baseline; the CO-210 packet has zero stale or rolling hits in `out/linear-34c13c4f-5839-4c64-b6c3-c2b30b7d7fe3/docs-freshness.json`.
- Follow-up `CO-218` tracks the separate stale pending launch-shell / ledger-repair decision-path risk after post-startup child-lane launch misses.

## Summary by CodeRabbit

* **New Features**
  * Child‑lane hydration: reserved placeholders are replaced by manifest‑backed status/progress when a matching started child run exists; progress events emitted on change.

* **Bug Fixes**
  * Fail‑closed handling for malformed or lineage‑mismatched manifests; historical ledger/proof records preserved; improved recorded‑at timestamp selection for provider telemetry.

* **Documentation**
  * Added PRD, tech spec, action plan, task checklist and registry entries describing behaviour and validation.

* **Tests**
  * Expanded coverage for hydration, refresh scopes, projection flows and edge cases.

Diff budget override: CO-210 needs the large worker-proof and selected-run projection diff plus focused regression coverage and required packet mirrors to keep the end-to-end child-lane hydration contract reviewable in one handoff.

### CODE REVIEW

File: orchestrator/src/cli/providerLinearWorkerRunner.ts [hunk 1]
```diff
@@ -4468,6 +4472,540 @@ export async function readProviderLinearWorkerChildLanes(
...
+  return priorProofChildLanes.find((prior) =>
+    prior.pipeline_id === existing.pipeline_id &&
+    prior.decision === existing.decision &&
+    prior.stream === existing.stream &&
+    prior.task_id === existing.task_id &&
+    prior.issue_id === existing.issue_id &&
+    prior.issue_identifier === existing.issue_identifier &&
+    prior.launched_at === existing.launched_at &&
+    normalizeOptionalString(prior.run_id)?.startsWith('launching-') !== true
+  ) ?? existing;
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply orchestrator/src/cli/providerLinearWorkerRunner.ts]: Fixed on top in `b1a0e4fde`.

The placeholder ledger row now stays in scan mode, so refreshes continue scanning `childCliDir` for matching manifests instead of pinning to the first hydrated proof row. I also added a regression that removes the first hydrated run dir and verifies we rediscover a replacement active child run while the ledger still preserves the `launching-*` reservation identity.

File: orchestrator/src/cli/control/selectedRunProjection.ts [hunk 2]
```diff
@@ -1227,6 +1241,90 @@ function shouldRefreshProviderLinearWorkerProjectionProof(
...
+    return (
+      pipelineId === PROVIDER_LINEAR_CHILD_LANE_PIPELINE_ID &&
+      decision === 'pending' &&
+      (
+        status === 'launching' ||
+        Boolean(runId?.startsWith('launching-')) ||
+        summary === PROVIDER_LINEAR_CHILD_LANE_RESERVED_SUMMARY
+      )
+    );
+  });
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply orchestrator/src/cli/control/selectedRunProjection.ts:1303]: @coderabbitai Fixed in `68ecd18f6`.

The projection refresh-scope helpers now fail closed on non-object child-lane entries, and I added a regression that seeds a malformed proof entry while still expecting the reserved placeholder to hydrate correctly instead of crashing. The same patch also hardens prior-proof child-lane matching so malformed entries cannot force a stale-proof fallback during refresh.

File: orchestrator/src/cli/providerLinearWorkerRunner.ts [hunk 3]
```diff
@@ -4468,6 +4472,540 @@ export async function readProviderLinearWorkerChildLanes(
...
+  const runId = normalizeOptionalString(parsed.run_id);
+  if (!runId) {
+    return null;
+  }
+  return {
+    runId,
+    issueId: normalizeOptionalString(parsed.issue_id),
+    issueIdentifier: normalizeOptionalString(parsed.issue_identifier),
+    workspacePath: normalizeOptionalString(parsed.workspace_path)
+  };
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply orchestrator/src/cli/providerLinearWorkerRunner.ts:4609]: @coderabbitai Fixed in `68ecd18f6`.

`readProviderLinearWorkerParentManifestHydrationMetadata()` now reads both snake_case and camelCase `run_id/runId`, `issue_id/issueId`, `issue_identifier/issueIdentifier`, and `workspace_path/workspacePath`, and I added a regression that hydrates a reserved child lane from a camelCase parent manifest.

File: orchestrator/src/cli/providerLinearWorkerRunner.ts [hunk 4]
```diff
@@ -4468,6 +4472,559 @@ export async function readProviderLinearWorkerChildLanes(
...
+  const manifestSummaryUpdatedAt =
+    normalizeOptionalString(manifest.summary) || normalizeOptionalString(manifest.status_detail)
+      ? normalizeOptionalString(manifest.updated_at)
+      : null;
+  return latestProviderLinearWorkerIsoTimestamp(
+    proofUpdatedAt,
+    normalizeOptionalString(manifest.completed_at),
+    manifestSummaryUpdatedAt,
+    startedAt
+  );
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply orchestrator/src/cli/providerLinearWorkerRunner.ts:5000]: @coderabbitai Fixed in `7b0965fa4`.

`providerLinearWorkerChildLaneSummaryRecordedAt()` now advances status-derived summaries from `updated_at`/`heartbeat_at` as well, and I added a focused regression covering a status-only queued child manifest so downstream freshness ordering tracks the visible summary change.

**Comment by Kbediako** [comment]
@coderabbitai Addressed the three live runtime review findings in `68ecd18f6` and `7b0965fa4`, and expanded the lineage mismatch regression matrix in the same pass. I’m intentionally leaving the remaining docs wording / shared test-fixture constant nits out of CO-210 because they are cosmetic and don’t change the child-lane hydration contract, runtime safety, or review-handoff readiness for this bugfix lane.
